src/member.rs

```diff
@@ -839,3 +839,113 @@ where
         let _ = network_exit.send(());
     }
 }
+
+#[cfg(test)]
```

Comment:
General comment/question, not directly related to the review: do we want to keep tests in the same files as the code? I see that this is nice in some ways (information is nicely concentrated in one place), but also any changes to tests mean the non-test parts of code also have to be recompiled (at least I think so, is that true?). This is not bothersome right now, but will probably be when the codebase and compilation times grow.

Reply:
This file is already quite big and this test is more of an integration test so I'll move it.